### DISCUSSION TITLE
Is there a way to prevent focus-stealing by new windows?

### DISCUSSION THREAD

**Discussion by winteriscariot** [labwc/labwc#2258] [Q&A]
I've noticed that if i'm working in an application and something triggers a new application to launch, the new application will grab my focus.

is there any way to prevent this, perhaps an open-in-background option that i've missed?

**Comment by Narrat** [comment]
Avoid focus stealing from new windows is kinda [on the ToDo](https://github.com/labwc/labwc/discussions/1987#discussioncomment-10065139).
Only preventing focus stealing from already open windows is currently possible.

**Reply by fungilife** [reply]
I had asked the same months ago and never got an answer.  No matter how you arrange options on focus on new windows focus still goes to an other window.  Even between two already open windows if takes off when a redraw is requested.
Example, work on a conky script from terminal, make a change, save, the focus leaves the terminal.

Very annoying, not critically :)

**Reply by Narrat** [reply]
Is it only a redraw from conky? Back then when I used it, I had the impression it would destroy and recreate the window on config reload. And there may still be something wonky with the wayland support of conky. I'm currently playing around with `eww` and there I didn't notice any focus stealing on changes in the config.
But for programs where I have `ignoreFocusRequest` in use it works like it should be (unless new windows are opened :D). So it is odd to read, that you cannot get it to work in general?

**Reply by fungilife** [reply]
No this is not conky specific, the openbox options for staying with your active window just don't work on labwc, as soon as an application is launched focus goes there by default.  Annoying but it may just be forcing a bad habit away?  Some people like it this way, but should be optional.

**Reply by Consolatis** [reply]
You could try `followMouse` and `followMouseRequiresMovement` (https://labwc.github.io/labwc-config.5.html#focus). However, that obviously focuses anything under the cursor.
